models/foreign_investor.rb

```diff
@@ -12,11 +12,11 @@ def initialize
   end
 
   def purchase_companies companies
-    company = companies.first
+    company = companies.shift
```

Comment:
i don't think this is correct, you shouldn't remove from companies because you may not buy it. the method buy_company will automatically remove the companies.

i think what you need to do though is to make sure companies is return the minimum priced one